### DISCUSSION TITLE
Building an older version

### DISCUSSION THREAD

**Discussion by ClockWorkAsylum** [smplayer-dev/smplayer#846] [General]
Hey all. As the title says, im trying to build an older version (19.10.2) in Ubuntu 23. I brought this up in a slightly different discussion (#826). The replacement mongoose.h file worked to finish the actual build. The problem now seems to be the create_deb doesnt  create the deb, it fails early in that process. if i use classic make all and make install, it seems to install, but nothing plays. does 19.10.2 only work with the versions of mpv or mplayer listed? my limited coding skills are at a loss.  any help is appreciated. thx

**Comment by smplayer-dev** [comment]
What error do you get when you try to create the deb package?

**Reply by ClockWorkAsylum** [reply]
it just stops after a few dh_ intructions. 

'dh_testdir
dh_testroot
dh_clean -k 
dh_clean: error: Compatibility levels before 7 are no longer supported (level 5 requested)
make: *** [debian/rules:56: install] Error 255
dh_clean: error: Compatibility levels before 7 are no longer supported (level 5 requested)'

again just copy/paste sorry.

**Comment by smplayer-dev** [comment]
Try changing the 5 in debian-orig/compat to 7.

**Reply by ClockWorkAsylum** [reply]
changing the compat file worked to complete the deb package. i changed the compat level to 10 because at 7 some of the files still show the 'below compat 10 is deprecated'.  ive tested both 7 and 10 builds. 

apt installs the deb just fine. 

trying to open and play any media causes smplayer to crash. ive verified mpv can easily play the test file.

here are the logs for mpv and smplayer. it looks to me like smplayer is not passing info and options to mpv or mpv is not reading from smplayer properly.

[smplayer log](https://github.com/smplayer-dev/smplayer/files/13512750/smplayer.log)
[mpv log](https://github.com/smplayer-dev/smplayer/files/13512751/mpv.log)

**Reply by ClockWorkAsylum** [reply]
EDIT:
i just tested if smplayer has the same crash with mplayer as with mpv.
it works acceptably using mplayer with no error, tho i would prefer mpv.
and again native mpv and mplayer both have no issue with the test video.

**Comment by smplayer-dev** [comment]
`Error parsing option input-file (option not found)`
That option was removed in mpv. So either you use an older version of mpv as well or use a newer version of smplayer.

**Reply by ClockWorkAsylum** [reply]
gotcha..  thank you for helping figure it out.
